### PR TITLE
Feature/update error handling

### PR DESCRIPTION
**Pull request by knees4bees**
## Is this a fix, feature, or refactor?
Fix and refactor

## What is the change?
* Refactors error handling to work with Router
* Adds testing for failed and not-ok network requests
* Minor cleanup

## What does it fix?
* Fixes problem with error handling where error message was not displaying for failed network request on movie detail page.

## Where should the reviewer start?
Most of the interesting changes are in `App.js` and `Error.js`.

## How should this be tested?
Fetch this branch and then 
```
npm i -D cypress
npm run test
```
Confirm that all tests are passing.

### CODE REVIEW

File: cypress/integration/rancid_tomatillos_spec.js [hunk 1]
```diff
@@ -16,6 +16,11 @@ describe('Rancid Tomatillos', () => {
       .should('have.class', 'nav-bar__search')
   });
 
+  it('Should not see a home button', () => {
+    cy.get('.nav-bar__home')
+      .should('not.be.visible')
+  });
+
```
Comment (itsnameissteven) [inline cypress/integration/rancid_tomatillos_spec.js:23]: nice catch

File: cypress/integration/rancid_tomatillos_spec.js [hunk 2]
```diff
@@ -84,11 +90,61 @@ describe('Movie details page', () => {
...
+      .get('.error__message')
+      .contains('Something went wrong')
+  });
 
-})
\ No newline at end of file
+  it('Should display error message if network down and clicked on movie poster', () => {
+      cy.intercept('https://rancid-tomatillos.herokuapp.com/api/v2/movies/632618', {
+        forceNetworkError: true
+      })
```
Comment (itsnameissteven) [inline cypress/integration/rancid_tomatillos_spec.js:143]: this is like a status code of 500?

File: src/components/App/App.js [hunk 3]
```diff
@@ -28,45 +27,21 @@ class App extends Component {
...
-        message = 'Invalid request.';
-        break;
-      default:
-        message = 'Oops! Something went wrong. Please try again.';
-    }
-
-    return (
-      <Error message={message} />
-    );
-  }
```
Comment (itsnameissteven) [inline src/components/App/App.js:58]: Great idea moving this into the error component. the file reads better now

File: src/components/Error/Error.js [hunk 4]
```diff
@@ -1,10 +1,23 @@
...
+    case 404:
+      message = 'Page not found.';
+      break;
+    case 500:
+      message = 'Oops! Something\'s broken on our end. Please try again later.';
+      break;
+    default:
+      message = 'Oops! Something went wrong. Please try again.';
+  }
+
```
Comment (itsnameissteven) [inline src/components/Error/Error.js:17]: nice refactor

**Review by itsnameissteven — APPROVED** [review]
Thanks for getting all that testing added. I think we are testing every possible outcome at the moment!

File: cypress/integration/rancid_tomatillos_spec.js (hunk 2 above)
Comment (knees4bees) [reply cypress/integration/rancid_tomatillos_spec.js:143]: No, I think it’s different. I think this is catching the case where the server is completely inaccessible, like if its power goes out. Not 100% on that though. I’d like to learn more about this in general.

**Pull request merged**